### PR TITLE
Add define option

### PR DESCRIPTION
**Pull request by developit**
Same as Terser's `--define`:

```
microbundle --define NODE_ENV=production
# this is also now set as the default ^
```

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -311,7 +312,16 @@ function createConfig(options, entry, format, writeMeta) {
 		return globals;
 	}, {});
 	if (options.globals && options.globals !== 'none') {
-		globals = Object.assign(globals, parseGlobals(options.globals));
+		globals = Object.assign(globals, parseMappingArgument(options.globals));
+	}
+
+	const NODE_ENV = process.env.NODE_ENV || 'production';
```
Comment (ForsakenHarmony) [inline src/index.js]: this is for libraries

Comment (ForsakenHarmony) [reply src/index.js]: so I don't support having that as the default at this point in the "toolchain"

Comment (jviide) [reply src/index.js]: That is a good point. `NODE_ENV` and `process.env.NODE_ENV` are probably best left undefined altogether by default.

Comment (developit) [reply src/index.js]: @ForsakenHarmony think it's better to not do inlining by default?  I'm good with that.

**Comment by developit** [comment]
Also apparently I'd managed to miss #233 when I opened this.

**Comment by weslleyaraujo** [comment]
Is there any plans for a release including this soon? 😄
that would be super helpful on the project I am working on 👍

**Comment by guicamillo** [comment]
> Is there any plans for a release including this soon? 😄
> that would be super helpful on the project I am working on 👍

Same thing over here :D

**Comment by developit** [comment]
yup

**Comment by ChrisBrownie55** [comment]
Since `--define` was released in version `0.10.1` it should probably be added to the README, seems like the only place it's documented currently is the releases tab.

**Pull request merged**